src/arco_era5/update_ar.py

```diff
@@ -0,0 +1,36 @@
+import apache_beam as beam
+import datetime
+import logging
+import xarray as xr
+import zarr
+
+from arco_era5 import HOURS_PER_DAY
+from dataclasses import dataclass
+from typing import Tuple
+
+logger = logging.getLogger(__name__)
+
+@dataclass
+class UpdateSlice(beam.PTransform):
+
+    target: str
+    init_date: str
+
+    def apply(self, offset_ds: Tuple[int, xr.Dataset, str]):
+        """Generate region slice and update zarr array directly"""
+        key, ds = offset_ds
```

Comment:
I think `ds` is actually a tuple of `(xr.Dataset, str)`. Is that intended?

Reply:
or, the type is incorrect.

Reply:
Yes the type is incorrect. Now I have updated that. Thanks for the suggestions.